lib/src/default_index/bit_set.rs

```diff
@@ -17,6 +17,72 @@ use std::cmp::max;
 use super::composite::CompositeCommitIndex;
 use super::entry::GlobalCommitPosition;
 
+/// Bit set of [`GlobalCommitPosition`]s.
+#[derive(Clone, Debug)]
+pub(super) struct PositionsBitSet {
+    data: Vec<u64>,
+}
+
+impl PositionsBitSet {
+    /// Creates bit set of the specified capacity.
+    pub fn with_capacity(len: u32) -> Self {
+        let bitset_len = usize::try_from(u32::div_ceil(len, u64::BITS)).unwrap();
+        PositionsBitSet {
+            data: vec![0; bitset_len], // request zeroed page
+        }
+    }
+
+    fn to_global_pos(&self, (bitset_pos, bit_pos): (u32, u32)) -> GlobalCommitPosition {
+        GlobalCommitPosition(bitset_pos * u64::BITS + bit_pos)
+    }
+
+    fn to_bitset_pos(&self, pos: GlobalCommitPosition) -> (u32, u32) {
+        (pos.0 / u64::BITS, pos.0 % u64::BITS)
+    }
+
+    /// Returns `true` if the given `pos` is set.
+    ///
+    /// Panics if the `pos` exceeds the capacity.
+    #[cfg_attr(not(test), expect(dead_code))] // TODO
+    pub fn get(&self, pos: GlobalCommitPosition) -> bool {
+        self.get_bit(self.to_bitset_pos(pos))
+    }
+
+    fn get_bit(&self, (bitset_pos, bit_pos): (u32, u32)) -> bool {
+        let bit = 1_u64 << bit_pos;
+        self.data[usize::try_from(bitset_pos).unwrap()] & bit != 0
```

Comment:
Can this be written as `self.get_set_bit((bitset_pos, bit_pos));` without performance impact? I would hope that the compiler would inline and simplify it but I don't know much about compilers. It's clearly not very complicated as is anyway, so I'm mostly just curious. If it does get optimized well, then we can also simplify the public interface to just have `get_set()`. Not a big deal either way if you don't feel like testing the performance impact, or if you prefer not to rely on what gets optimized out.

Reply:
No, `.get_set_bit()` needs `&mut self`. Maybe we can instead merge `.get_set*()` and `.set*()` (or delete `.get_set*()` at all), but I didn't feel strongly and just duplicated the code.